### PR TITLE
[3.8.] Avatar documentation

### PR DESCRIPTION
**Pull request by ntacheva**

### CODE REVIEW

File: components/avatar/overview.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+The <a href="https://www.telerik.com/blazor-ui/avatar" target="_blank">Blazor Avatar component</a> is typically used to display images, icons or initials representing people or other entities.
+
+It also gives you the flexibility to customize its size, color, fill mode and more.
+
+## Creating Avatar
+
+1. Use the `TelerikAvatar` tag to add the component to your razor page.
+
+1. Declare your desired content inside the `TelerikAvatar` tag - text, icon or image.
```
Comment (svdimitr) [inline components/avatar/overview.md:21]: We can add the actual technical term for the space between the opening and closing tag - `ChildContent`.

File: components/avatar/overview.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+## Appearance
+
+The Avatar provides multiple options for configuring its look. [Explore the available appearance settings...]({%slug avatar-appearance%})
+
+## Parameters
+
+The Blazor Avatar provides the following parameters that allow you to configure the component:
+
+@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)
```
Comment (svdimitr) [inline components/avatar/overview.md:50]: This template is no longer needed. Dimo made improvements to automatically add these styles to the page.

File: components/avatar/overview.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+| `Bordered` | `bool` | Whether the Avatar will have border. |
+| `Type` | `AvatarType` <br/> (`Image`) | The type of the Avatar. Read more in the [Types article]({%slug avatar-types%}). |
+
+### Styling and Appearance
+
+The following parameters enable you to customize the appearance of the Blazor Avatar:
+
+| Parameter | Type | Description |
+| --- | --- | --- |
+| `Class` | `string` | The CSS class that will be rendered on the main wrapping element of the Avatar. (`<div class="k-avatar>`) Use it to [override the theme or apply custom styles]({%slug themes-override%}). |
```
Comment (svdimitr) [inline components/avatar/overview.md:63]: | `Class` | `string` | Add a custom CSS class to the  `<div class="k-avatar>`. Use it to [override the theme or apply custom styles]({%slug themes-override%}). |

File: components/avatar/types.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+The Avatar serves as a wrapper - you should declare the desired content inside the `<TelerikAvatar>` tag and specify the `Type` based on the content type.
+
+The `Type` parameter accepts a member of the `AvatarType` enum:
+
+* `Image` - default value
+* `Text`
+* `Icon`
+
+We recommend setting the correct `Type` to the Avatar depending on the content you want to display inside it. The Avatar will apply different styles to the content based on the specified `Type`.
```
Comment (svdimitr) [inline components/avatar/types.md:23]: maybe use a `>tip`? 

You can move that sentence after the next paragraph.

File: components/avatar/types.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+tags: telerik,blazor,avatar,type,image,icon,text
+published: True
+position: 5
+---
+
+# Avatar Types
+
+The `Type` parameter of the Avatar for Blazor allows you to specify what kind of content will be visualized in the component.
+
+The Avatar serves as a wrapper - you should declare the desired content inside the `<TelerikAvatar>` tag and specify the `Type` based on the content type.
```
Comment (svdimitr) [inline components/avatar/types.md:15]: Mention the `ChildContent` here.

File: components/avatar/overview.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+position: 0
+---
+
+# Blazor Avatar Overview
+
+The <a href="https://www.telerik.com/blazor-ui/avatar" target="_blank">Blazor Avatar component</a> is typically used to display images, icons or initials representing people or other entities.
+
+It also gives you the flexibility to customize its size, color, fill mode and more.
+
+## Creating Avatar
```
Comment (dimodi) [inline components/avatar/overview.md:17]: SEO

```suggestion
## Creating Blazor Avatar
```

File: components/avatar/overview.md (hunk 1 above)
Comment (dimodi) [reply components/avatar/overview.md:21]: Not sure how recognized is `ChildContent` by devs, but I agree that I stumbled upon this step and my first thought was "how"? Probably I expected a property and didn't understand that we are suggesting nested content inside the component tag.

File: components/avatar/overview.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+1. Use the `TelerikAvatar` tag to add the component to your razor page.
+
+1. Declare your desired content inside the `TelerikAvatar` tag - text, icon or image.
+
+1. Set the [Avatar type]({%slug avatar-types%}) based on your content.
```
Comment (dimodi) [inline components/avatar/overview.md:23]: This step is not actionable unless the reader refers to other docs content. In such cases, we should either add one more sentence for clarification, or just suggest a specific setting for now and leave the other options for later.

File: components/avatar/overview.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+## Types
+
+The Avatar `Type` defines the type of the content that the component will display - it may be text, icon or image. [Read more about the specifics of the Avatar types...]({%slug avatar-types%})
+
+## Appearance
+
+The Avatar provides multiple options for configuring its look. [Explore the available appearance settings...]({%slug avatar-appearance%})
+
+## Parameters
```
Comment (dimodi) [inline components/avatar/overview.md:46]:
```suggestion
## Avatar Parameters
```

File: components/avatar/overview.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+The Blazor Avatar provides the following parameters that allow you to configure the component:
+
+@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)
+
+| Parameter    | Type  | Description |
+| ----------- | ----------- | -------|
+| `Bordered` | `bool` | Whether the Avatar will have border. |
+| `Type` | `AvatarType` <br/> (`Image`) | The type of the Avatar. Read more in the [Types article]({%slug avatar-types%}). |
+
+### Styling and Appearance
```
Comment (dimodi) [inline components/avatar/overview.md:57]: The number of parameters is small, so why split them into groups? Moreover, `Bordered` is for appearance as well, but it's in the "main" group.

File: components/avatar/types.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+description: Explore the Blazor Avatar Types.
+slug: avatar-types
+tags: telerik,blazor,avatar,type,image,icon,text
+published: True
+position: 5
+---
+
+# Avatar Types
+
+The `Type` parameter of the Avatar for Blazor allows you to specify what kind of content will be visualized in the component.
```
Comment (dimodi) [inline components/avatar/types.md:13]:
```suggestion
The `Type` parameter of the Avatar for Blazor defines what kind of content will render inside the component. The parameter allows the Avatar to adjust its styling to its content.
```

File: components/avatar/appearance.md [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+description: Appearance settings of the Avatar for Blazor.
+slug: avatar-appearance
+tags: telerik,blazor,avatar,appearance
+published: True
+position: 10
+---
+
+# Appearance Settings
+
+You can control the appearance of the Avatar by setting the following attribute:
```
Comment (dimodi) [inline components/avatar/appearance.md:13]: In all cases - plural.

```suggestion
You can control the appearance of the Avatar by setting the following parameters:
```

File: components/avatar/appearance.md [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+
+You can increase or decrease the size of the Avatar by setting the `Size` parameter to a member of the `Telerik.Blazor.ThemeConstants.Avatar.Size` class:
+
+| Class members | Manual declarations |
+|---------------|--------|
+| `Small`   |`sm`|
+| `Medium` (default value)  |`md`|
+| `Large`   |`lg`|
+
+> The default value of the `Size` will take precedence over the values of the `Width` and `Height` parameters. Set the `Size` to an empty `string` to apply your custom `Width` and `Height`.
```
Comment (dimodi) [inline components/avatar/appearance.md:32]: I would challenge the developer for this decision. Why it's necessary?

File: components/avatar/appearance.md [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+            JD
+        </TelerikAvatar>
+        </div>
+    }
+}
+````
+
+## Rounded
+
+The `Rounded` parameter applies the `border-radius` CSS rule to the Avatar to achieve curving of the edges. You can set it to a member of the `Telerik.Blazor.ThemeConstants.Avatar.Rounded` class:
```
Comment (dimodi) [inline components/avatar/appearance.md:58]: CSS rule is the whole block together with the selectors.

```suggestion
The `Rounded` parameter applies the `border-radius` CSS style to the Avatar to achieve curving of the edges. You can set it to a member of the `Telerik.Blazor.ThemeConstants.Avatar.Rounded` class:
```

File: components/avatar/appearance.md [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+                JD
+            </TelerikAvatar>
+        </div>
+    }
+}
+````
+
+## FillMode
+
+The `FillMode` controls how the TelerikAvatar is filled. You can set it to a member of the `Telerik.Blazor.ThemeConstants.Avatar.FillMode` class:
```
Comment (dimodi) [inline components/avatar/appearance.md:93]: We are using the word "fill" to explain "fillmode". It's not that self-evident. Try something else, e.g. "background", etc.

File: components/avatar/types.md [hunk 15]
```diff
@@ -0,0 +1,51 @@
+---
+title: Types
```
Comment (dimodi) [inline components/avatar/types.md:2]: Cross link the Types and Appearance articles via Next Steps

File: components/avatar/overview.md (hunk 1 above)
Comment (ntacheva) [reply components/avatar/overview.md:21]: Can you please clarify to make sure we are on teh same page? Your suggestion is

```suggestion
1. Declare your desired `ChildContent` inside the `TelerikAvatar` tag - text, icon or image.
```

File: components/avatar/overview.md (hunk 3 above)
Comment (ntacheva) [reply components/avatar/overview.md:63]: This suggestion contradicts with the practice we've incorporated for the descriptions. The description should start with a noun and not a verb.

File: components/avatar/overview.md (hunk 9 above)
Comment (ntacheva) [reply components/avatar/overview.md:57]: I wanted to keep the structure for dividing the parameters as in other components (e.g. [AutoComplete](https://docs.telerik.com/blazor-ui/components/autocomplete/overview#parameters)). However, here the number is indeed pretty small, so I will combine the tables.

File: components/avatar/overview.md (hunk 3 above)
Comment (svdimitr) [reply components/avatar/overview.md:55]: Ok, so how about - `A custom CSS class to the .....`?

File: components/avatar/appearance.md (hunk 12 above)
Comment (ntacheva) [reply components/avatar/appearance.md:32]: To be discussed with the development team.

File: components/avatar/overview.md [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+
+The <a href="https://www.telerik.com/blazor-ui/avatar" target="_blank">Blazor Avatar component</a> is typically used to display images, icons or initials representing people or other entities.
+
+You can set different [Avatar types]({%slug avatar-types%}) and customize its [size]({%slug avatar-appearance%}#size), [fill mode]({%slug avatar-appearance%}#fillmode) and more.
+
+## Creating Blazor Avatar
+
+1. Use the `TelerikAvatar` tag to add the component to your razor page.
+
+1. Declare your desired content (as `ChildContent`) inside the `TelerikAvatar` tag - text, icon or image. This example demonstrates Avatar with initials(text).
```
Comment (dimodi) [inline components/avatar/overview.md:21]:
```suggestion
1. Declare your desired content (as `ChildContent`) inside the `TelerikAvatar` tag - text, icon or image. This example demonstrates Avatar with initials (text).
```

File: components/avatar/overview.md [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+
+1. Set the [Avatar type]({%slug avatar-types%}) to `Text`.
```
Comment (dimodi) [inline components/avatar/overview.md:23]:
```suggestion
1. Set the [Avatar type]({%slug avatar-types%}). In this case, we are using `Text`.
```

File: components/avatar/types.md [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+
+The Avatar serves as a wrapper - you should declare the desired content as `ChildContent` inside the `<TelerikAvatar>` tag and specify the `Type` based on the content type.
+
+The `Type` parameter accepts a member of the `AvatarType` enum:
+
+* `Image` - default value
+* `Text`
+* `Icon`
+
+If the `Type` parameter value is not matching the type of the content, you will not get the optimal appearance of the component. For example, using `AvatarType.Image` when you render text in the component will result in misaligned, not centered text.
```
Comment (dimodi) [inline components/avatar/types.md:23]:
```suggestion
If the `Type` parameter value is not matching the type of the content, you will not get the optimal appearance of the component. For example, using `AvatarType.Image` when you render text in the component will result in misaligned text.
```

**Review by dimodi — APPROVED** [review]

File: components/avatar/types.md [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+
+>tip We recommend setting the correct `Type` to the Avatar depending on the content you want to display inside it. The Avatar will apply different styles to the content based on the specified `Type`.
```
[the remark this answers is not in the document]
Comment (ntacheva) [reply components/avatar/types.md:25]: Sort of. I added it as a tip with a recommendation for better visibility.

**Pull request merged**